webapp/channels/src/components/advanced_text_editor/advanced_text_editor.test.tsx

```diff
@@ -0,0 +1,326 @@
+                    roles: 'user_roles',
+                    locale: 'en',
+                },
+            },
+            statuses: {
+                current_user_id: 'online',
+            },
+        },
+        roles: {
+            roles: {
+                user_roles: {permissions: []},
+                channel_roles: {permissions: []},
+                team_roles: {permissions: []},
+            },
+        },
+        groups: {
+            groups: {},
+        },
+        emojis: {
+            customEmoji: {},
+        },
+        preferences: {
+            myPreferences: {},
+        },
+    },
+    websocket: {
+        connectionId: 'connection_id',
+    },
+};
+
+const emptyDraft: PostDraft = {
+    message: '',
+    uploadsInProgress: [],
+    fileInfos: [],
+    channelId,
+    rootId: '',
+    createAt: 0,
+    updateAt: 0,
+};
+
+const baseProps = {
+    location: 'CENTER',
+    message: '',
+    showEmojiPicker: false,
+    uploadsProgressPercent: {},
+    currentChannel: initialState.entities.channels.channels.current_channel_id as Channel,
+    channelId,
+    postId: '',
+    errorClass: null,
+    serverError: null,
+    postError: null,
+    isFormattingBarHidden: false,
+    draft: emptyDraft,
+    badConnection: false,
+    handleSubmit: jest.fn(),
+    removePreview: jest.fn(),
+    showSendTutorialTip: false,
+    setShowPreview: jest.fn(),
+    shouldShowPreview: false,
+    maxPostSize: 100,
+    canPost: true,
+    applyMarkdown: jest.fn(),
+    useChannelMentions: false,
+    currentChannelTeammateUsername: '',
+    currentUserId,
+    canUploadFiles: true,
+    enableEmojiPicker: true,
+    enableGifPicker: true,
+    handleBlur: jest.fn(),
+    handlePostError: jest.fn(),
+    emitTypingEvent: jest.fn(),
+    handleMouseUpKeyUp: jest.fn(),
+    postMsgKeyPress: jest.fn(),
+    handleChange: jest.fn(),
+    toggleEmojiPicker: jest.fn(),
+    handleGifClick: jest.fn(),
+    handleEmojiClick: jest.fn(),
+    hideEmojiPicker: jest.fn(),
+    toggleAdvanceTextEditor: jest.fn(),
+    handleUploadProgress: jest.fn(),
+    handleUploadError: jest.fn(),
+    handleFileUploadComplete: jest.fn(),
+    handleUploadStart: jest.fn(),
+    handleFileUploadChange: jest.fn(),
+    getFileUploadTarget: jest.fn(),
+    fileUploadRef: React.createRef<FileUpload>(),
+    prefillMessage: jest.fn(),
+    textboxRef: React.createRef<Textbox>(),
+    isThreadView: false,
+    ctrlSend: true,
+    codeBlockOnCtrlEnter: true,
+    onMessageChange: jest.fn(),
+    onEditLatestPost: jest.fn(),
+    loadPrevMessage: jest.fn(),
+    loadNextMessage: jest.fn(),
+    replyToLastPost: jest.fn(),
+    caretPosition: 0,
+};
+
+describe('components/avanced_text_editor/advanced_text_editor', () => {
+    describe('keyDown behavior', () => {
+        it('Enter should call postMsgKeyPress', () => {
+            const postMsgKeyPress = jest.fn();
+            renderWithRealStore(
+                <WebSocketContext.Provider value={WebSocketClient}>
```

Comment:
If you're able to switch these to `renderWithFullContext`, this should probably be added to the context that that adds